lib/jxl/encode.cc

```diff
@@ -870,6 +882,7 @@ JxlEncoderStatus JxlEncoderAddJPEGFrame(const JxlEncoderOptions* frame_settings,
   }
   queued_frame->frame.SetFromImage(std::move(*io.Main().color()),
                                    io.Main().c_current());
+  // TODO: maybe add extra channels? if JPEG-1 recompression allows to add any.
```

Comment:
The spec allows adding alpha or other extra channels to a recompressed jpeg - from the Part 1 point of view, there is no difference between 'jxl' VarDCT and 'recompressed-jpeg' VarDCT.

From the Part 2 point of view (where `jbrd` reconstruction is described), it's also no problem: extra channels will simply be ignored when reconstructing the JPEG.

From an end-user perspective, it is maybe a bit confusing to have jxl files that can be reconstructed to a jpeg but the jpeg is actually only part of the jxl (e.g. the alpha channel gets stripped in the process). But I suppose that's a matter of djxl and other applications to e.g. show a warning in such a case indicating that there's more info in the jxl than what is represented in the reconstructed jpeg.

Reply:
Agreed, clarified TODO